File: XWeather/XWeather/Extensions/DateExtensions.cs
using System;

namespace XWeather
{
	public static class DateExtensions
	{
		public static bool IsBetween (this DateTime date, DateTime start, DateTime end) => start < date && date < end;

		public static bool IsOutside (this DateTime date, DateTime start, DateTime end) => start > date || date > end;
	}
}

File: XWeather-Swift/XWeather/Extensions/DateExtensions.swift
//
//  DateExtensions.swift
//  XWeather
//
//  Created by Colby Williams on 6/11/17.
//  Copyright © 2017 Colby Williams. All rights reserved.
//

import Foundation

extension Date {
	
	func isBetween (_ start: Date, end: Date) -> Bool {
		return start < self && self < end
	}
	
	func isOutside (_ start: Date, end: Date) -> Bool {
		return start > self || self > end
	}
}
